File: Subtree in N ary Tree[1-May-2023]/Solution.java
// >> JAVA CODE <<

class Solution {
    HashMap<String, Integer> map = new HashMap<String, Integer>();
    
    String solve(Node root) {
        if(root == null) {
            return "";
        }
        
        String returnValue = Integer.toString(root.data);
        
        for(Node child : root.children) {
            returnValue += "#" + solve(child);
        }
        
        map.put(returnValue, map.getOrDefault(returnValue, 0) + 1);
        
        return returnValue;
    }
    
    int duplicateSubtreeNaryTree(Node root) {
        solve(root);
        
        int ans = 0;
        for(int val : map.values()) {
            if(val > 1) {
                ans++;
            }
        }
        
        return ans;
    }
}

File: Subtree in N ary Tree[1-May-2023]/solution.py
#   >> PYTHON CODE <<


def duplicateSubtreeNaryTree(self, root):
        self.map={}
        
        def solve(root):
            if(root is None):
                return ""
            
            returnValue=root.__str__()
            
            for ch in root.children:
                returnValue=returnValue+"#"+solve(ch)
            
            
            self.map[returnValue]=self.map.get(returnValue,0)+1
            
            return returnValue
        
        
        solve(root)
        
        ans=0
        for val in self.map.values():
            if(val>1):
                ans+=1
        
        return ans